atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueService.java

```diff
@@ -0,0 +1,337 @@
+        }
+    }
+
+    @Override
+    public Map<Cell, Long> getLatestTimestamps(TableReference tableRef,
+            Map<Cell, Long> timestampByCell) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getLatestTimestamps({}, {} cells)",
+                tableRef, timestampByCell.size())) {
+            return delegate().getLatestTimestamps(tableRef, timestampByCell);
+        }
+    }
+
+    @Override
+    public byte[] getMetadataForTable(TableReference tableRef) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getMetadataForTable({})", tableRef)) {
+            return delegate().getMetadataForTable(tableRef);
+        }
+    }
+
+    @Override
+    public Map<TableReference, byte[]> getMetadataForTables() {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getMetadataForTables()")) {
+            return delegate().getMetadataForTables();
+        }
+    }
+
+    @Override
+    public ClosableIterator<RowResult<Value>> getRange(TableReference tableRef,
+            RangeRequest rangeRequest,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getRange({}, {}, {})",
+                tableRef, rangeRequest, timestamp)) {
+            return delegate().getRange(tableRef, rangeRequest, timestamp);
+        }
+    }
+
+    @Override
+    public ClosableIterator<RowResult<Set<Long>>> getRangeOfTimestamps(TableReference tableRef,
+            RangeRequest rangeRequest,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getRangeOfTimestamps({}, {}, {})",
+                tableRef, rangeRequest, timestamp)) {
+            return delegate().getRangeOfTimestamps(tableRef, rangeRequest, timestamp);
+        }
+    }
+
+    @Override
+    public Map<Cell, Value> getRows(TableReference tableRef,
+            Iterable<byte[]> rows,
+            ColumnSelection columnSelection,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getRows({}, {} rows, {}, {})",
+                tableRef, Iterables.size(rows), columnSelection, timestamp)) {
+            return delegate().getRows(tableRef, rows, columnSelection, timestamp);
+        }
+    }
+
+    @Override
+    public Map<byte[], RowColumnRangeIterator> getRowsColumnRange(TableReference tableRef,
+            Iterable<byte[]> rows,
+            BatchColumnRangeSelection columnRangeSelection,
+            long timestamp) {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("getRowsColumnRange({}, {} rows, {}, {})",
```

Comment:
This could potentially leak data as the column ranges of dynamic columns could contain information at the product level.